core/trino-spi/src/main/java/io/trino/spi/exchange/ExchangeContext.java

```diff
@@ -13,41 +13,16 @@
  */
 package io.trino.spi.exchange;
 
+import io.opentelemetry.api.trace.Span;
 import io.trino.spi.Experimental;
 import io.trino.spi.QueryId;
 
-import java.util.StringJoiner;
-
-import static java.util.Objects.requireNonNull;
-
 @Experimental(eta = "2023-09-01")
-public class ExchangeContext
+public interface ExchangeContext
```

Comment:
Based on the current usages, we could have leave ExchangeContext as a concrete class.
What's the reason to make it an interface?

Reply:
Adding the Span ends up affecting all the OpenTelemetry dependencies changing them from runtime to provided. That ends up rippling into all the other pom files. It makes the PR much larger.

Reply:
I don't follow

Reply:
When adding a span to ExchangeContext, it creates a new dependency on opentelemetry-context. The build error is:
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project trino-spi: Compilation failure
[ERROR] /Users/walter.weiss/git/opensource/os_trino/core/trino-spi/src/main/java/io/trino/spi/exchange/ExchangeContext.java:[33,27] cannot access io.opentelemetry.context.ImplicitContextKeyed
[ERROR]   class file for io.opentelemetry.context.ImplicitContextKeyed not found

To get around this, I changed the dependency scope from runtime to provided. However, that forces a change in scope for nearly all the other projects as well, so to minimize the blast radius it seemed more straightforward to change ExchangeContext to an interface. If you have an alternative suggestion, I am open to it.